### PR TITLE
Enable support for Python 3.9+

### PR DESCRIPTION
**Pull request by satta**
This commit basically removes Python 3.10+ language features from the code required to use the `python` operator in pipelines. This mostly means rewriting code using:

* PEP-604 (Union types written with `|`)
* PEP-613 (Explicit Type Aliases)
* PEP-634 (Structural Pattern Matching)

We also lower the version requirement in `pyproject.toml` and update dependencies in `poetry.lock`.

Tests still succeed:
```
❯ pyenv exec poetry run pytest
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.11.2, pytest-8.1.1, pluggy-1.4.0
rootdir: /home/satta/tmp/tenzir/python
configfile: pyproject.toml
plugins: asyncio-0.23.6
asyncio: mode=Mode.AUTO
collected 15 items / 1 skipped                                                                                                                                            

tests/test_convert.py ..                                                                                                                                            [ 13%]
tests/test_utils_arrow.py .............                                                                                                                             [100%]

====================================================================== 15 passed, 1 skipped in 0.25s ======================================================================
```

If there is anything else to test that you know of, please do so. I have no knowledge of what remains unconvered by pytest.

### CODE REVIEW

File: libtenzir/aux/simdjson
Comment (dominiklohmann) [inline libtenzir/aux/simdjson:1]: I assume this was bumped by accident.

**Review by dominiklohmann — COMMENTED** [review]
I am willing to accept this, but please be aware that we will not restrict ourselves to compatibility with Python 3.9 in the future, as this slows down developers. If as part of future development of the Tenzir Python library someone wants to bump the minimum version, we will allow it.

File: python/pyproject.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ documentation = "https://tenzir.com/docs"
 repository = "https://github.com/tenzir/tenzir"
 
 [tool.poetry.dependencies]
-python = "^3.10"
+python = "^3.9"
```
[the remark this answers is not in the document]
Comment (satta) [reply python/pyproject.toml:26]: Will change that. Thanks

File: libtenzir/aux/simdjson
Comment (satta) [reply libtenzir/aux/simdjson:1]: It was, will push an update excluding this.

File: python/poetry.lock [hunk 2]
```diff
@@ -1,9 +1,10 @@
-# This file is automatically @generated by Poetry 1.7.1 and should not be changed by hand.
+# This file is automatically @generated by Poetry and should not be changed by hand.
 
 [[package]]
 name = "aiohttp"
 version = "3.9.3"
 description = "Async http client/server framework (asyncio)"
+category = "main"
```
[the remark this answers is not in the document]
Comment (satta) [reply python/poetry.lock:7]: Didn't do this explicitly, `poetry` did that. https://github.com/python-poetry/poetry/issues/5702#issuecomment-1139512812 seems to imply these categories are probably obsolete.

**Comment by satta** [comment]
> I am willing to accept this, but please be aware that we will not restrict ourselves to compatibility with Python 3.9 in the future, as this slows down developers. If as part of future development of the Tenzir Python library someone wants to bump the minimum version, we will allow it.

TBH once Threat Bus is gone we will quickly update to more recent versions of everything anyway. So I don't worry we will be kept back by this for much longer.

**Pull request merged**